src/utils/marks/marks.js

```diff
@@ -92,10 +92,28 @@ export class Mark {
         if (!this.range) {
             return [];
         }
+
         const rects = Array.from(this.range.getClientRects());
         const stringRects = rects.map((r) => JSON.stringify(r));
-        const setRects = new Set(stringRects);
-        return Array.from(setRects).map((sr) => JSON.parse(sr));
+        const stringRectsSet = new Set(stringRects);
+        const rectsSet = Array.from(stringRectsSet).map((sr) => JSON.parse(sr));
+
+        let filteredRects = [];
+        for (let i = 0; i < rectsSet.length; i++) {
```

Comment:
Since we are creating an array from this for loop, I think we could do this with a `reduce` method?

```js
return rectSet.reduce((filteredRects, curRect) => {
  const shouldNotPush = rectsSet.some((item) => curRect !== item && contains(curRect, item));
  if (!shouldNotPush) {
    filteredRects.push(curRect);
  }
  return filteredRects;
}, []);
```

Reply:
good call, updated!